### PR TITLE
Refactoring and simplifying

### PR DESCRIPTION
**Pull request by alilee**
Really pulling things apart.

### CODE REVIEW

File: src/input.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+impl bevy::prelude::Plugin for Plugin {
+    fn build(&self, app: &mut App) {
+        app.add_system(update_directions);
+    }
+}
+
+fn update_directions(mut query: Query<&mut snake::Moving, (With<Location>, With<snake::Head>)>, mut event_keyboard: EventReader<KeyboardInput>) {
+    for mut moving in query.iter_mut() {
+        for event in event_keyboard.iter() {
```
Comment (alilee) [inline src/input.rs:18]: This lets you turn around if you are quick enough!

File: src/render.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    Vec3::new(
+        calculate_pixel_coordinate(x, board::WIDTH),
+        calculate_pixel_coordinate(y, board::HEIGHT),
+        height,
+    )
+}
+
+fn calculate_pixel_coordinate(coord: i8, size: i8) -> f32 {
+    let coord = coord as f32 * TILE_SIZE;
+    if size % 2 == 1 {
```
Comment (alilee) [inline src/render.rs:60]: These are some rubbish changes I made here - sorry.

File: src/snake.rs [hunk 3]
```diff
@@ -25,6 +21,7 @@ fn setup(mut commands: Commands) {
         .insert(Head)
         .insert(Location::default())
         .insert(Moving::default())
+        // FIXME: this bundle should be in render (somehow)!
```
Comment (alilee) [inline src/snake.rs]: This sprite bundle should really be part of render, but you need to associate it with the entity when the snake is spawned.

File: src/input.rs (hunk 1 above)
Comment (interacsion) [reply src/input.rs:18]: That's why I had the `Moving` component remember the last direction as well...

File: src/snake.rs (hunk 3 above)
Comment (interacsion) [reply src/snake.rs]: we could make another system for that, but that'll require creating a new stage. I'd use a utility method.

Comment (alilee) [reply src/snake.rs]: With bevy, Is there some way to make a trigger on entity creation? ie. `render` plugin could watch for `snake::Head` creations and add the bundle.

File: src/input.rs (hunk 1 above)
Comment (alilee) [reply src/input.rs:18]: Aha! So what is the desired behaviour here? Should we let the player override their previous choice inside a tick, or one input per tick? Also, I'm thinking ahead to where one snake is played on WASD and another on IJKL and another on arrows...

File: src/snake.rs (hunk 3 above)
Comment (alilee) [reply src/snake.rs]: Ok did this with Added<snake::Head>, will be 1 frame late, but not a problem.

File: src/input.rs (hunk 1 above)
Comment (interacsion) [reply src/input.rs:18]: I think the player should be able to override his choice between ticks, but he shouldn't be able to go backwards.

File: src/snake.rs (hunk 3 above)
Comment (interacsion) [reply src/snake.rs]: That's clever! I didn't know that `Added<>` was a thing. I tried to do that exact thing with a startup system that's `.after(snake::setup)`, and queries the entity `With<snake::Head>`, but the entity wouldn't have spawned in the same stage.

File: src/render.rs (hunk 2 above)
Comment (interacsion) [reply src/render.rs:60]: look at #5

**Pull request merged**